src/staking/block_validator.cpp

```diff
@@ -120,78 +123,76 @@ class BlockValidatorImpl : public AbstractBlockValidator {
   //!
   //! This signature is checked here against the public key which is used to unlock the
   //! stake. The piece of information which is signed is the block hash.
-  void CheckBlockSignatureInternal(const CBlock &block, BlockValidationResult &result) const {
+  BlockValidationResult CheckBlockSignatureInternal(const CBlock &block) const {
     const uint256 block_hash = block.GetHash();
 
     const std::vector<CPubKey> keys = staking::ExtractBlockSigningKeys(block);
     if (keys.empty()) {
-      result.AddError(Error::INVALID_BLOCK_PUBLIC_KEY);
-      return;
+      return BlockValidationResult(Error::INVALID_BLOCK_PUBLIC_KEY);
     }
     for (const CPubKey &pubkey : keys) {
       if (pubkey.Verify(block_hash, block.signature)) {
-        return;
+        return BlockValidationResult();
```

Comment:
should it be `BlockValidationResult::success` as in other cases above?

Reply:
It should. Will change.

Reply:
@kostyantyn Fixed in https://github.com/dtr-org/unit-e/pull/1098/commits/75fada74f6801242bc1f6674eead1b01452ece10

I also grepped for `BlockValidationResult()` to make sure there are no other leftovers.